src/run/run.c

```diff
@@ -394,6 +400,28 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_wait = true;
                         break;
 
+                case ARG_WORKING_DIRECTORY:
+                        r = parse_path_argument_and_warn(optarg, true, &arg_working_directory);
+                        if (r < 0)
+                                return r;
+
+                        break;
+
+                case 'd': {
+                        _cleanup_free_ char *p = NULL;
+
+                        r = safe_getcwd(&p);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to get current working directory: %m");
+
+                        if (empty_or_root(p))
+                                arg_working_directory = mfree(arg_working_directory);
+                        else
+                                free_and_strdup(&arg_working_directory, p);
```

Comment:
```C
free_and_strdup(&arg_working_directory, empty_or_root(p) ? NULL : p);
```

Reply:
uh, this was supposed to use free_and_replace() actually, but I fucked that up (I blame emacs abbrev mode...). I'll correct that, but when I do, the second argument needs to be an lvalue so that it can be assigned NULL...

Reply:
(this missed an OOM check even in the old version...)